### PR TITLE
Add asin, acos, tan, atan operators

### PR DESCRIPTION
**Pull request by highker**

### CODE REVIEW

File: caffe2/python/gradient_check_test.py [hunk 1]
```diff
@@ -269,6 +269,28 @@ def testCos(self):
                 res, grad, grad_estimated = checker.CheckSimple(op, [X], 0, [0])
                 self.assertTrue(res)
 
+
+class TestAcos(test_util.TestCase):
```
Comment (salexspb) [inline caffe2/python/gradient_check_test.py]: don't add new tests here, use folder caffe2/python/operator_test, create new files for your operators (or group of operators). Then use helpers such as assertReferenceChecks and assertGradientChecks. Fbgs those for more examples

File: caffe2/python/hypothesis_test.py [hunk 2]
```diff
@@ -1041,6 +1041,26 @@ def cos_ref(input_tensor):
             inputs=[input_tensor],
             reference=cos_ref)
 
+    @given(input_tensor=hu.arrays(
```
Comment (salexspb) [inline caffe2/python/hypothesis_test.py]: ditto, don't add stuff to this file, create new files

**Review by salexspb — CHANGES_REQUESTED** [review]

**Comment by highker** [comment]
Comments addressed. Thanks @salexspb

File: caffe2/python/operator_test/asin_op_test.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+from __future__ import absolute_import
```
Comment (salexspb) [inline caffe2/python/operator_test/asin_op_test.py]: looks like asin and acos can go together and reuse the same test with except passing different lambda for reference as well as different operator name, what do you think?

File: caffe2/python/operator_test/tan_op_test.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from __future__ import absolute_import
```
Comment (salexspb) [inline caffe2/python/operator_test/tan_op_test.py]: and other same tests as well :)

File: caffe2/operators/asin_op.cc [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+OPERATOR_SCHEMA(AsinGradient).NumInputs(2).NumOutputs(1).IdenticalTypeAndShape();
+
+class GetAsinGradient : public GradientMakerBase {
+  using GradientMakerBase::GradientMakerBase;
+  vector<OperatorDef> GetGradientDefs() override {
+    Argument one_arg;
+    one_arg.set_name("value");
+    one_arg.set_f(1);
+    return vector<OperatorDef>{CreateOperatorDef(
```
Comment (salexspb) [inline caffe2/operators/asin_op.cc]: how about just fusing this manually? this implementation on Caffe2 graph seems to be  inefficient. And we are not there for full compilers support yet.

File: caffe2/operators/atan_op.cc [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+OPERATOR_SCHEMA(AtanGradient).NumInputs(2).NumOutputs(1).IdenticalTypeAndShape();
+
+class GetAtanGradient : public GradientMakerBase {
+  using GradientMakerBase::GradientMakerBase;
+  vector<OperatorDef> GetGradientDefs() override {
+    Argument one_arg;
+    one_arg.set_name("value");
+    one_arg.set_f(1);
+    return vector<OperatorDef>{CreateOperatorDef(
```
Comment (salexspb) [inline caffe2/operators/atan_op.cc]: ditto

File: caffe2/operators/acos_op.cc [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+OPERATOR_SCHEMA(AcosGradient).NumInputs(2).NumOutputs(1).IdenticalTypeAndShape();
+
+class GetAcosGradient : public GradientMakerBase {
+  using GradientMakerBase::GradientMakerBase;
+  vector<OperatorDef> GetGradientDefs() override {
+    Argument one_arg;
+    one_arg.set_name("value");
+    one_arg.set_f(1);
+    return vector<OperatorDef>{CreateOperatorDef(
```
Comment (salexspb) [inline caffe2/operators/acos_op.cc]: ditto

**Review by salexspb — CHANGES_REQUESTED** [review]

**Review by salexspb — APPROVED** [review]
should be good to go, thanks!

**Pull request merged**